### DISCUSSION TITLE
(Dialog) ButtonClose and ButtonMaximizable emit when click them?

### DISCUSSION THREAD

**Discussion by JoseAntonioNavasSolutia** [primefaces/community#514] [PrimeVue]
I want them to emit an event when the button is clicked

**Comment by FlipWarthog** [comment]
Dialog emits events for hide, after-hide, maximize, unmaximize: https://primevue.org/dialog/#api.dialog.emits

**Comment by philgruneich** [comment]
Regarding this, can we cancel/prevent the hiding of the dialog with a ConfirmDialog, for example?

**Comment by tugcekucukoglu** [comment]
There is `closable` property whose default value is `true`. https://primevue.org/dialog/#api.dialog.props.closable

**Reply by philgruneich** [reply]
That would remove the close button. Is there a way to have it while the closable prop is set to false so we can intercept the closure of the dialog?

**Reply by ataylor32** [reply]
@philgruneich I opened a new discussion about this specific thing: https://github.com/orgs/primefaces/discussions/3289
